### PR TITLE
Trim async debugging when debugging feature is disabled

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml [hunk 1]
```diff
@@ -15,5 +15,8 @@
     <type fullname="System.Resources.ResourceReader">
       <method signature="System.Boolean InitializeBinaryFormatter()" body="stub" value="false" feature="System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization" featurevalue="false" />
     </type>
+    <type fullname="System.Threading.Tasks.Task" feature="System.Diagnostics.Debugger.IsSupported" featurevalue="false">
+      <field name="s_asyncDebuggingEnabled" value="false" initialize="false" />
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml:19]: What does initialize=false do?

**Review by eerhardt — APPROVED** [review]
I think this looks good. What kind of savings are you seeing with this change?

File: src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml (hunk 1 above)
Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml:19]: It tells the linker not to generate field initialization code

**Comment by marek-safar** [comment]
About 4k but it unblocks trimming for more even source once we clean them up as well

**Pull request merged**